### DISCUSSION TITLE
使用自己的数据集训练版面分析有差异

### DISCUSSION THREAD

**Discussion by beetter** [PaddlePaddle/PaddleOCR#12315] [Q&A]
请提供下述完整信息以便快速定位问题/Please provide the following information to quickly locate the problem
- 系统环境/System Environment：
ubuntu 22.04
cuda 11.8
- 版本号/Version：Paddle：  PaddleOCR： 问题相关组件/Related components：
PaddleDetection-2.6.0
使用picodet_lcnet_x1_0_fgd_layout_cdla训练模型上训练自己的数据集
- 运行指令/Command Code：
export CUDA_VISIBLE_DEVICES=0,1
python -m paddle.distributed.launch --gpus '0,1'  tools/train.py \
    -c configs/picodet/legacy_model/application/layout_analysis/picodet_lcnet_x1_0_layout.yml \
    --eval
- 完整报错/Complete Error Message：
<img width="999" alt="WechatIMG791" src="https://github.com/PaddlePaddle/PaddleOCR/assets/49197005/a47c6ba4-3862-4aa5-bc42-d6e7f2ac5f43">
使用coco格式的数据集，自己标注了5000张，4000张训练集，1000张测试集，训练后的模型识别效果总是有些漏识别、多识别、识别检测框不全的问题，请问怎么解决。
请尽量不要包含图片在问题中/Please try to not include the image in the issue.

**Comment by TingquanGao** [comment]
想问下为什么要fine-tune呢，是我们提供的预训练模型效果较差吗？fine-tune训练的超参数是用的默认配置吗？

**Comment by beetter** [comment]
> 想问下为什么要fine-tune呢，是我们提供的预训练模型效果较差吗？fine-tune训练的超参数是用的默认配置吗？

测试了效果不太好，我的排版没有顺序，有些会漏识别或者错识别。训练的超参是默认的configs/picodet/legacy_model/application/layout_analysis/picodet_lcnet_x1_0_layout.yml文件，使用labelme标注的多边形数据，使用x2coco.py转换成训练集和测试集，yml文件配置如下：
_BASE_: [
  '../../../../runtime.yml',
  '../../_base_/picodet_esnet.yml',
  '../../_base_/optimizer_100e.yml',
  '../../_base_/picodet_640_reader.yml',
]

pretrain_weights: https://paddledet.bj.bcebos.com/models/pretrained/LCNet_x1_0_pretrained.pdparams
weights: output/picodet_lcnet_x1_0_layout/model_final
find_unused_parameters: True
use_ema: true
cycle_epoch: 10
snapshot_epoch: 1
epoch: 100

PicoDet:
  backbone: LCNet
  neck: CSPPAN
  head: PicoHead

LCNet:
  scale: 1.0
  feature_maps: [3, 4, 5]

metric: COCO
num_classes: 15

TrainDataset:
  !COCODataSet
    image_dir: train
    anno_path: annotations/instance_train.json
    dataset_dir: ./dataset/publaynet/cocome311/
    data_fields: ['image', 'gt_bbox', 'gt_class', 'is_crowd']

EvalDataset:
  !COCODataSet
    image_dir: val
    anno_path: annotations/instance_val.json
    dataset_dir: ./dataset/publaynet/cocome311/

TestDataset:
  !ImageFolder
    anno_path: ./dataset/publaynet/cocome311/annotations/instance_val.json


worker_num: 8
eval_height: &eval_height 800
eval_width: &eval_width 608
eval_size: &eval_size [*eval_height, *eval_width]

TrainReader:
  sample_transforms:
  - Decode: {}
  - RandomCrop: {}
  - RandomFlip: {prob: 0.5}
  - RandomDistort: {}
  batch_transforms:
  - BatchRandomResize: {target_size: [[768, 576], [800, 608], [832, 640]], random_size: True, random_interp: True, keep_ratio: False}
  - NormalizeImage: {is_scale: true, mean: [0.485,0.456,0.406], std: [0.229, 0.224,0.225]}
  - Permute: {}
  batch_size: 24
  shuffle: true
  drop_last: true
  collate_batch: false

EvalReader:
  sample_transforms:
  - Decode: {}
  - Resize: {interp: 2, target_size: [800, 608], keep_ratio: False}
  - NormalizeImage: {is_scale: true, mean: [0.485,0.456,0.406], std: [0.229, 0.224,0.225]}
  - Permute: {}
  batch_transforms:
  - PadBatch: {pad_to_stride: 32}
  batch_size: 24
  shuffle: false


TestReader:
  inputs_def:
    image_shape: [1, 3, 800, 608]
  sample_transforms:
  - Decode: {}
  - Resize: {interp: 2, target_size: [800, 608], keep_ratio: False}
  - NormalizeImage: {is_scale: true, mean: [0.485,0.456,0.406], std: [0.229, 0.224,0.225]}
  - Permute: {}
  batch_transforms:
  - PadBatch: {pad_to_stride: 32}
  batch_size: 24
  shuffle: false
只更改了数据集地址。

**Comment by beetter** [comment]
还有就是我使用如下命令导出推理模型时，可以导出模型，并且在output_inference里面生成了4个文件
python tools/export_model.py \
-c configs/picodet/legacy_model/application/layout_analysis/picodet_lcnet_x1_0_layout.yml \
-o weights='output/picodet_lcnet_x1_0_layout/best_model.pdparams'  \
--output_dir=output_inference
当我把推理模型和paddleOCR结合后就出现了下面这个错误
<img width="1016" alt="WechatIMG792" src="https://github.com/PaddlePaddle/PaddleOCR/assets/49197005/b7884b3e-becb-4ea0-b6bd-e371f2beb743">

**Comment by sralvins** [comment]
> 还有就是我使用如下命令导出推理模型时，可以导出模型，并且在output_inference里面生成了4个文件 python tools/export_model.py -c configs/picodet/legacy_model/application/layout_analysis/picodet_lcnet_x1_0_layout.yml -o weights='output/picodet_lcnet_x1_0_layout/best_model.pdparams' --output_dir=output_inference 当我把推理模型和paddleOCR结合后就出现了下面这个错误 <img alt="WechatIMG792" width="1016" src="https://private-user-images.githubusercontent.com/49197005/312799114-b7884b3e-becb-4ea0-b6bd-e371f2beb743.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA5OTExMTQsIm5iZiI6MTcxMDk5MDgxNCwicGF0aCI6Ii80OTE5NzAwNS8zMTI3OTkxMTQtYjc4ODRiM2UtYmVjYi00ZWEwLWI2YmQtZTM3MWYyYmViNzQzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzIxVDAzMTMzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZmOWI3MDI5ZWFjODA4OGMwZTQ0NWI2Zjg1ZWYxZTIyYTA5NTNjZWMzYTA4ZGQ5ZGNiZjgzOTY0YTQxYzM2ZDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.kRmyvMN8mY_4Yn8NJpRxuHPsy0n9qJ8t8vRf6bDHbQ0">

got same errors

**Comment by TingquanGao** [comment]
想问下导出的inference模型文件直接推理测试会报错吗？以及，可以将该文件提供给我们用于排查问题吗？

**Comment by sralvins** [comment]
> 想问下导出的inference模型文件直接推理测试会报错吗？以及，可以将该文件提供给我们用于排查问题吗？

[picodet_lcnet_x1_0_layout.zip](https://github.com/PaddlePaddle/PaddleOCR/files/14691324/picodet_lcnet_x1_0_layout.zip)

**Comment by liyuweihuo** [comment]
请问有最终答复吗？自己数据再fine-tune效果不太好

**Comment by xuwinrar** [comment]
同问
